widget/embedded/src/components/ConfirmWalletsModal/ConfirmWalletsModal.tsx

```diff
@@ -156,62 +131,52 @@ export function ConfirmWalletsModal(props: PropTypes) {
           (!isWalletRequiredFor(chain) && !customDestination))
     );
 
-  const isAddressMatched =
-    !!customDestination &&
-    showCustomDestination &&
-    lastStepToBlockchain &&
-    !isValidAddress(lastStepToBlockchain, customDestination);
-
-  const resetCustomDestination = () => {
-    setShowCustomDestination(false);
-    setCustomDestination(null);
-    setSelectableWallets((selectableWallets) => {
-      let anyWalletSelected = false;
-      return selectableWallets.map((selectableWallet) => {
-        if (
-          !anyWalletSelected &&
-          selectableWallet.chain === lastStepToBlockchain?.name
-        ) {
-          anyWalletSelected = true;
-          return {
-            ...selectableWallet,
-            selected: true,
-          };
-        }
-        return selectableWallet;
-      });
-    });
-  };
-
-  const handleClearCustomDestination = () => setCustomDestination('');
-
-  const handlePasteCustomDestination = async (
-    event: MouseEvent<HTMLButtonElement>
+  const updateSelectableWallets = (
+    wallets: ConnectedWallet[],
+    chainName: string,
+    updateWallet: (wallet: ConnectedWallet) => ConnectedWallet
   ) => {
-    event.preventDefault();
-    if (navigator.clipboard !== undefined) {
-      const pastedText = await navigator.clipboard.readText();
-      setCustomDestination(pastedText);
-      customDestinationInputRef?.current?.focus();
-    }
+    return wallets.map((wallet) => {
+      if (wallet.chain === chainName) {
+        return updateWallet(wallet);
+      }
+      return wallet;
+    });
   };
-
   const handleCustomDestinationCollapsibleOpenChange = (checked: boolean) => {
+    setCustomDestinationOpen(checked);
     if (!checked) {
-      resetCustomDestination();
+      setSelectableWallets((selectableWallets) => {
+        let anyWalletSelected = false;
```

Comment:
I don't get this. `anyWalletSelected` is initialized with `false` and checked if it is `false` set it `true` and never used it again.

Reply:
This is for  the case we may be connected to multiple wallets from the same blockchain and just need to choose one of them. This code is meant to avoid this issue.